pkg/datamover/testutils.go

```diff
@@ -0,0 +1,438 @@
+				RepoPassword:   "",
+				ConfigFilePath: kopiacmd.DefaultConfigFilePath,
+				LogDirectory:   kopiacmd.DefaultLogDirectory,
+			},
+			ServerAddress:    "https://0.0.0.0:51515",
+			TLSCertFile:      testTLSCertPath,
+			TLSKeyFile:       testTLSKeyPath,
+			ServerUsername:   testKopiaRepoServerAdminUsername,
+			ServerPassword:   testKopiaRepoServerAdminPassword,
+			AutoGenerateCert: false,
+			Background:       true,
+		})
+
+	stdout, stderr, err := kube.Exec(cli, namespace, pod.GetName(), pod.Spec.Containers[0].Name, cmd, nil)
+	format.Log(pod.GetName(), pod.Spec.Containers[0].Name, stdout)
+	format.Log(pod.GetName(), pod.Spec.Containers[0].Name, stderr)
+	if err != nil {
+		return errors.Wrap(err, "Failed to start Kopia API server")
+	}
+	return nil
+}
+
+func getServerStatusCommand(ctx context.Context, cli kubernetes.Interface, namespace string, tlsSecret *corev1.Secret) ([]string, error) {
+	fingerprint, err := kopia.ExtractFingerprintFromCertSecret(ctx, cli, tlsSecret.GetName(), namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error Extracting Fingerprint from the TLS Certificates")
+	}
+
+	cmd := kopiacmd.ServerStatus(
+		kopiacmd.ServerStatusCommandArgs{
+			CommandArgs: &kopiacmd.CommandArgs{
+				RepoPassword:   "",
+				ConfigFilePath: kopiacmd.DefaultConfigFilePath,
+				LogDirectory:   kopiacmd.DefaultLogDirectory,
+			},
+			ServerAddress:  "https://0.0.0.0:51515",
+			ServerUsername: testKopiaRepoServerAdminUsername,
+			ServerPassword: testKopiaRepoServerAdminPassword,
+			Fingerprint:    fingerprint,
+		})
+	return cmd, nil
+}
+
+func waitForServerReady(ctx context.Context, cli kubernetes.Interface, namespace string, pod *corev1.Pod, tlsSecret *corev1.Secret) error {
+	cmd, err := getServerStatusCommand(ctx, cli, namespace, tlsSecret)
+	if err != nil {
+		return errors.Wrap(err, "Error Getting Server Status Command")
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, defaultServerStartTimeout)
+	defer cancel()
+
+	return repositoryserver.WaitTillCommandSucceed(ctx, cli, cmd, namespace, pod.GetName(), pod.Spec.Containers[0].Name)
+}
+
+func addTestUserInKopiaRepositoryServer(cli kubernetes.Interface, namespace string, pod *corev1.Pod) error {
+	testUser := fmt.Sprintf("%s@%s", testKopiaRepoServerUsername, defaultKopiaRepositoryHost)
+	cmd := kopiacmd.ServerAddUser(
+		kopiacmd.ServerAddUserCommandArgs{
+			CommandArgs: &kopiacmd.CommandArgs{
+				RepoPassword:   defaultKopiaRepositoryPassword,
+				ConfigFilePath: kopiacmd.DefaultConfigFilePath,
+				LogDirectory:   kopiacmd.DefaultLogDirectory,
+			},
+			NewUsername:  testUser,
+			UserPassword: testKopiaRepoServerAdminPassword,
+		})
+	stdout, stderr, err := kube.Exec(cli, namespace, pod.GetName(), pod.Spec.Containers[0].Name, cmd, nil)
+	format.Log(pod.GetName(), pod.Spec.Containers[0].Name, stdout)
+	format.Log(pod.GetName(), pod.Spec.Containers[0].Name, stderr)
+	if err != nil {
+		return errors.Wrap(err, "Failed to add users in Kopia API server")
```

Comment:
What happens if the user already exists? Do we not want to set password for existing user rather that adding the user in that case?

Reply:
Since the test is going to run once and cleanup itself, I think it will be of no use to check if user already exists.

Reply:
And Users are also random everytime